pallets/automation-time/src/fees.rs

```diff
@@ -75,30 +76,36 @@ where
 	OU: OnUnbalanced<NegativeImbalanceOf<T>>,
 {
 	/// Ensure the fee can be paid.
-	fn can_pay_fee(who: &T::AccountId, fee: BalanceOf<T>) -> Result<(), DispatchError> {
+	fn can_pay_fee(&self) -> Result<(), DispatchError> {
+		let fee = self.execution_fee.saturating_add(self.xcmp_fee);
+
 		if fee.is_zero() {
 			return Ok(())
 		}
 
-		let free_balance = T::Currency::free_balance(who);
-		let new_amount =
-			free_balance.checked_sub(&fee).ok_or(DispatchError::Token(BelowMinimum))?;
-		T::Currency::ensure_can_withdraw(who, fee, WithdrawReasons::FEE, new_amount)?;
+		// Manually check for ExistenceRequirement since MultiCurrency doesn't currently support it
+		let free_balance = T::MultiCurrency::free_balance(self.currency_id, &self.owner);
+		free_balance
+			.checked_sub(&fee)
+			.ok_or(DispatchError::Token(BelowMinimum))?
+			.checked_sub(&T::MultiCurrency::minimum_balance(self.currency_id))
+			.ok_or(DispatchError::Token(BelowMinimum))?;
+		T::MultiCurrency::ensure_can_withdraw(self.currency_id.into(), &self.owner, fee)?;
 
 		Ok(())
 	}
 
 	/// Withdraw the fee.
-	fn withdraw_fee(who: &T::AccountId, fee: BalanceOf<T>) -> Result<(), DispatchError> {
+	fn withdraw_fee(&self) -> Result<(), DispatchError> {
+		let fee = self.execution_fee.saturating_add(self.xcmp_fee);
+
 		if fee.is_zero() {
 			return Ok(())
 		}
 
-		let withdraw_reason = WithdrawReasons::FEE;
-
-		match T::Currency::withdraw(who, fee, withdraw_reason, ExistenceRequirement::KeepAlive) {
-			Ok(imbalance) => {
-				OU::on_unbalanceds(Some(imbalance).into_iter());
+		match T::MultiCurrency::withdraw(self.currency_id, &self.owner, fee) {
+			Ok(_) => {
+				// TODO: fees need to go to multi currency enabled treasury
```

Comment:
This is a big `TODO`, but this pr was already getting big so I opted to separate this functionality. However, this is a blocker for release. Without implementing a treasury that can accept foreign currencies all automation fees will get burned.

Reply:
XcmFees in particular need to be held in a holding account, not burned.

Reply:
Yes, noted and I agree.